components/script/dom/htmlinputelement.rs

```diff
@@ -1143,7 +1144,25 @@ impl VirtualMethods for HTMLInputElement {
 
 impl FormControl for HTMLInputElement {}
 
-impl Validatable for HTMLInputElement {}
+impl Validatable for HTMLInputElement {
+    fn as_element_validatable(&self) -> &Element {
+        self.upcast()
+    }
+    fn is_instance_validatable(&self) -> bool {
```

Comment:
This method likely needs to implement the various cases where input elements are [barred from constraint validation](https://html.spec.whatwg.org/multipage/forms.html#barred-from-constraint-validation) (click the bold text to see a list of places where the term is used).